src/filter/QueenSacrificeFilter.ts

```diff
@@ -0,0 +1,106 @@
+import { VideoFilter } from "@/model/VideoFilter";
+import { Video } from "@/model/Video";
+import { Pgn } from "@/model/Pgn";
+import { usePgnsStore } from "@/stores/pgnsStore";
+
+/* 
+  There are some cases where a queen sacrifice may be wrongly detected:
+  - A blunder where the queen is captured without any compensation. However, it is unlikely that such games would be analyzed on the channel.
+  - A queen exchange combination, such as a check in-between move before recapturing the other queen.
+
+  There are also some cases where the queen sacrifice may not be detected:
+  - This function does not currently handle the scenario in which a queen exchange takes place and, later, a pawn is promoted to a queen, and then sacrificed.
+*/
+function isQueenSacrifice(pgn: string): boolean {
+  const moves = pgn
+    .replace(/\d+\.\s+/g, "") // Remove move numbers
+    .trim()
+    .split(/\s+/); // Split by spaces
+
+  const captureRegex = /([QRNB])?([a-h])?x([a-h][1-8])?/;
+
+  let whiteQueenPosition = "d1"; // White Queen starts at d1
+  let blackQueenPosition = "d8"; // Black Queen starts at d8
+
+  const lastMoveIndex = moves.length - 1;
+
+  for (let i = 0; i < moves.length; i++) {
+    const move = moves[i];
+
+    const captureMatch = move.match(captureRegex);
+
+    // If it's a capture move
+    if (captureMatch) {
+      const targetSquare = captureMatch[3]; // The square where the capture happens
+
+      const isWhiteQueenCapture = targetSquare === whiteQueenPosition;
+      const isBlackQueenCapture = targetSquare === blackQueenPosition;
+
+      if (isWhiteQueenCapture) {
```

Comment:
I think you can easliy get rid of duplication in conditional branches by doing something like this

```
if (isWhiteQueenCapture || isBlackQueenCapture) {
  var otherQueenPosition = isWhiteQueenCapture ? blackQueenPosition : whiteQueenPosition
  if (captureMatch[i] === "Q") {
    otherQueenPosition = targetSquare
  }
  const nextMove = moves[i + 1];
  const nextCaptureMatch = nextMove?.match(captureRegex);
  const nextTargetSquare = nextCaptureMatch?.[3];
  const nextMoveIsQueenCapture = nextTargetSquare === otherQueenPosition;
  return !nextMoveIsQueenCapture && i !== lastMoveIndex;
}
```